internal/dataplane/sendconfig/error_handling_test.go

```diff
@@ -0,0 +1,161 @@
+package sendconfig
+
+import (
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseFlatEntityErrors(t *testing.T) {
+	log := logrus.New()
+	tests := []struct {
+		name    string
+		body    []byte
+		want    []ResourceError
+		wantErr bool
+	}{
+		{
+			name: "a route nested under a service, with two and one errors, respectively",
+			want: []ResourceError{
+				{
+					Name:       "httpbin",
+					Namespace:  "67338dc2-31fd-47b6-85a9-9c11d347d090",
+					Kind:       "Ingress",
+					APIVersion: "networking.k8s.io/v1",
+					UID:        "ea569579-f7e9-4d4e-973b-b207bfb848d8",
+					Problems: map[string]string{
+						"methods": "cannot set methods when protocols is grpc or grpcs",
+					},
+				},
+				{
+					Name:       "httpbin",
+					Namespace:  "67338dc2-31fd-47b6-85a9-9c11d347d090",
+					Kind:       "Service",
+					APIVersion: "v1",
+					UID:        "e7e5c93e-4d56-4cc3-8f4f-ff1fcbe95eb2",
+					Problems: map[string]string{
+						"":     "failed conditional validation given value of field protocol",
```

Comment:
It's some odd artifact of how the Kong error system works that I can't quite explain. This kinda applies to `path` also, since it must be null because of the value of `protocol`, but the way the Lua validation works applies that explanation to the entire entity. That in turn results in an empty key in the new error system.

That's at least my best guess looking at something like https://github.com/Kong/kong/blob/19fff8038517896c3d08da63020b9cf7e775ab83/spec/02-integration/03-db/02-db_core_entities_spec.lua#L1578-L1579 in the gateway unit tests. Dunno why these ones end up like that whereas the `methods` failure mentions the relation just in the `methods` error.

Reply:
Copying some additional context from @flrgh re the Kong internals for posterity:

To explain a little bit more about why things are weird, you're seeing several different layers of schema validation:

1. field-level validation is used for things that more or less be validated statically without any entity context and flattens to a `type: "field"` error => `entity.attr must be an integer`
2. entity-level validation is used for dynamic logic that hinges on multiple values and flattens to a `type: "entity"` error => `if entity.attr > 0 then entity.other must be null`
3. subschema validation, which can include additional field-level or entity-level validation

Subschemas are a way of telling the framework to conditionally include and apply additional schema based on the value of some entity attribute. They serve a similar purpose as entity-level validators but are much more modular in nature.
The most prevalent example of this is plugins. Every plugin has a "base" schema  with standard fields (name, protocols, service, route, consumer, etc), but the value of `plugin.name` tells the framework `locate the schema for plugin.name and apply it to plugin.config`, thereby allowing each kong plugin to define its own config schema in a separate file.
This is why you are seeing a field-level error for `route.methods` even though it's technically a composite-ish error that requires evaluating multiple fields. The route entity uses subschemas, and the grpc subschema has a field-level validator for `route.methods`:
https://github.com/Kong/kong/blob/19fff8038517896c3d08da63020b9cf7e775ab83/kong/db/schema/entities/routes_subschemas.lua#L51